src/cijoe/cli/cli.py

```diff
@@ -367,13 +367,23 @@ def cli_workflow(args):
     return err
 
 
-def create_adhoc_workflow(args, paths):
-    paths = list(map(Path, paths))
-    if len(set(path.stem for path in paths)) != len(paths):
+def create_adhoc_workflow(args, steps: list[str]):
+    parent_dirs = set()
+
+    def normalise(step: str):
+        if step.endswith(".py"):
+            path = Path(step)
+            parent_dirs.add(path.parent)
+            return path.stem
+        else:
+            return step
+
+    steps = list(map(normalise, steps))
+    if len(set(steps)) != len(steps):
         log.error("Duplicate script file names not allowed.")
         sys.exit(1)
 
-    resources = get_resources([step.parent for step in paths])
+    resources = get_resources(list(parent_dirs))
```

Comment:
When the user invokes a command like cijoe core.repository_prep foo/bar.py --config .., the order of execution is significant. For example, a repository may need to be prepared before running a build, install, or deploy script. However, using set(), which is unordered, may unintentionally reorder the list of scripts to run.

Reply:
The set is only used to add the local scripts' parent directories to the search paths of the resource collector. The order of the steps in remained in the `steps` list :)

Reply:
Ohhh, awesome then!